serde_arrow/src/arrow_impl/api.rs

```diff
@@ -76,17 +73,14 @@ impl ArrowBuilder {
     /// given fields.
     ///
     pub fn new(fields: &[Field]) -> Result<Self> {
-        let fields = fields
-            .iter()
-            .map(GenericField::try_from)
-            .collect::<Result<Vec<_>>>()?;
-        Ok(Self(generic::GenericBuilder::new_for_arrays(&fields)?))
+        let schema = SerdeArrowSchema::from_arrow_fields(fields)?;
+        Ok(Self(ArrayBuilder::new(&schema)?))
     }
 
     /// Add a single record to the arrays
     ///
     pub fn push<T: Serialize + ?Sized>(&mut self, item: &T) -> Result<()> {
```

Comment:
- Re `slice::from_ref`: I would like to keep the type `?Sized` as there is no reason for it to be `Sized`. 
- Re for loop: note that extend does not expect any particular sequence types, but any type `T` that serializes to a sequence. Hence I don't think think it's possible to use a for loop.

I think the best option would be to not create a generic `ArrayBuilder` for the outer sequence, but a `ListBuilder` directly. This way it would be possible to access the item builder directly. Since [`ArrayBuilder::new`](https://github.com/chmp/serde_arrow/blob/e48662dd5a87a98de8d4264215bc24ce8f71af7d/serde_arrow/src/internal/serialization_ng/array_builder.rs#L91) anyway constructs a `ListBuilder` that should be easy enough to change.